rclpy/test/test_parameters_callback.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from rcl_interfaces.msg import SetParametersResult
+import rclpy
+from rclpy.parameter import Parameter
+
+
+class TestParametersCallback(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        rclpy.init()
+
+    @classmethod
+    def tearDownClass(cls):
+        rclpy.shutdown()
+
+    def setUp(self):
+        self.node = rclpy.create_node('parameters_callback_node')
+
+    def tearDown(self):
+        self.node.destroy_node()
+
+    def test_set_callback_accepting_all(self):
+        global callback_called
```

Comment:
is it necessary for this to be global? could it be local to each test case and nonlocal to the callbacks?

Reply:
TIL about `nonlocal` That's precisely what I needed. Thanks.

Reply:
resolved by 73f6cf2